Давлетова Алия

davletova opened the pull request.

davletova committed:
add solution

Zhigalov commented on phoneBook.js:
```diff
@@ -1,52 +1,43 @@
+		var regexpEmail = /^([a-z0-9_-]+\.)*[a-z0-9_-]+@[а-яa-z0-9_-]+(\.[а-яa-z0-9_-]+)*\.[а-яa-z]{2,6}$/;
+		
+		if (regexpPhone.test(phone)) {
+			if (regexpEmail.test(email)) {
```
Если одно условие лежит внутри другого, то можно записать условия через `&&`

Zhigalov commented on phoneBook.js:
```diff
@@ -1,52 +1,43 @@
 
-var phoneBook; // Здесь вы храните записи как хотите
+var phoneBook = []; 
+var person = {};
```
А для чего здесь эта переменная? Почему бы не объявить её максимально близко к месту использования?

Zhigalov commented on phoneBook.js:
```diff
@@ -1,52 +1,43 @@
-
+		
+		var regexpPhone = /^(\+\d{1,3}|\d{1,3})?\s*(\(\d{3}\)|\d{3})\s*\d{3}(\-\d\-|\d)\d{3}$/;
+		var regexpEmail = /^([a-z0-9_-]+\.)*[a-z0-9_-]+@[а-яa-z0-9_-]+(\.[а-яa-z0-9_-]+)*\.[а-яa-z]{2,6}$/;
```
В js есть проблема с перечислением `а-я`, туда не попадает буква `ё`. Придется её добавить отдельно

Zhigalov commented on phoneBook.js:
```diff
@@ -1,52 +1,43 @@
+    for (var i = 0; i < phoneBook.length; i++) {
+		person = phoneBook[i];
+		if (person.name.indexOf(query) >=0 || person.phone.indexOf(query) >=0 || person.email.indexOf(query) >=0) {
+			console.log(person.name+', '+person.phone+', '+person.email);
```
Телефон хочется всегда выводить одинаково отформатированным. Вне зависимости от формата ввода.
То есть всегда вот таким: `+7 (999) 666-7-778`
Мы наверное это плохо сформулировали в задании

Zhigalov commented on phoneBook.js:
```diff
@@ -1,52 +1,43 @@
-   Функция поиска записи в телефонную книгу.
-   Поиск ведется по всем полям.
-*/
 module.exports.find = function find(query) {
```
Если я ничего не передам в качестве запроса, то хочу увидеть все записи,
сейчас ты ни одной записи не показываешь, если вызвать `.find();`

Zhigalov commented on phoneBook.js:
```diff
@@ -1,52 +1,43 @@
+		person = phoneBook[i];
+		if (person.name.indexOf(query) >=0 || person.phone.indexOf(query) >=0 || person.email.indexOf(query) >=0) {
+			console.log('Удален контакт:'+person.name+', '+person.phone+', '+person.email);
+			phoneBook.splice(i,1);
```
:+1:

Zhigalov commented on phoneBook.js:
```diff
@@ -1,52 +1,43 @@
-    // Ваша невероятная магия здесь
-
+		
+		var regexpPhone = /^(\+\d{1,3}|\d{1,3})?\s*(\(\d{3}\)|\d{3})\s*\d{3}(\-\d\-|\d)\d{3}$/;
```
Вот такой телефон `+7 (111) 777 2 222` не пролез

Zhigalov commented:
Хорошо! Нужно доделать замечания, в челом все ок!
Попробуй сделать задание на звездочку, мне кажется у тебя должно получиться :-)

davletova committed:
Задача поправлена

У меня возник вопрос по срокам сдачи задач: в слайдах к одной из лекций было указано, что на задачу дается 2 недели, за эти 2 недели необходимо прислать решение(первый вариант, до каких-либо правок) или же задача должна быть зачтена за этот срок?
И если же оветом на заданный вопрос является второй вариант, и мне теперь необходимо прийти в яндекс для сдачи, когда можно это сделать?

Zhigalov commented:
К сожалению окончательные правки пришли только после дедлайна проверки, поэтому могу поставить только половинку балла :disappointed:

Zhigalov commented:
И не забывай пожалуйста писать коммент после того как сделала коммит, чтобы я сразу пришел и перепроверил
